src/dmd/expressionsem.d

```diff
@@ -3587,36 +3587,58 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         }
         else if (tb.isscalar())
         {
+            Identifier id;
+            if (tb.isZeroInit(exp.loc) == true)
+            {
+                id = Identifier.idPool("___d_newitemT");
+            }
+            else
+            {
+                id = Identifier.idPool("___d_newitemiT");
+            }
+            auto tiargs = new Objects();
+            tiargs.push(exp.newtype);
+            auto ti = new TemplateInstance(exp.loc, id, tiargs);
+
+            Expression newExpLowering = new ScopeExp(exp.loc, ti);
+            auto arguments = new Expressions();
+
+            newExpLowering = new CallExp(exp.loc, newExpLowering, arguments);
+            newExpLowering = newExpLowering.expressionSemantic(sc);
+
             if (!nargs)
             {
             }
             else if (nargs == 1)
             {
                 Expression e = (*exp.arguments)[0];
                 e = e.implicitCastTo(sc, tb);
-                (*exp.arguments)[0] = e;
+                newExpLowering = new ConstructExp(newExpLowering.loc, newExpLowering, e.addressOf());
+                newExpLowering.expressionSemantic(sc);
             }
```

Comment:
```
newExpLowering = newExpLowering.expressionSemantic(sc);
```

Reply:
You're also calling expressionSemantic() twice on the same expression.

Reply:
Even this works:

```d
newExpLowering = new ConstructExp(newExpLowering.loc, newExpLowering, e.addressOf()).expressionSemantic(sc);
```

But it might be a bit too long.